### ISSUE TITLE
Newton diagnostics fails with unkown error at runtime (due to uncaught exceptions?)

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#10115]
The `NewtonDiagnostics.ThermoFluidSystem.M3`, as well as `M4`, `M5`, and `M6` from [NewtonDiagnostics.zip](https://github.com/OpenModelica/OpenModelica/files/10518553/NewtonDiagnostics.zip) fails at runtime with error `0xcffffffff` (unknown error?) when using version 2dfc4ff from PR #9263.

My educated guess is that this could be due to an uncatched error when trying to compute `sqrt(p_o - p_d)` or `sqrt(p_s - p_i)` with a negative argument. Potentially invalid operations (division, square root, logarithm) are handled by the C runtime in a way that allows to cleanly abort the computation and retry with different guess values or step size, see the generated NewtonDiagnostics.ThermoFluidSystem.M3_06inz.c source code file. 

We need to figure out how to integrate this smoothly with @TeusvanderStelt's Newton diagnostics code.

@AnHeuermann, @kabdelhak, Teus's code calls the generated code to compute the residuals of nonlinear algebraic systems. When computing them, exceptions may happen, e.g. negative square root, division by zero, log of negative number, non-integer power of negative number, etc. In all these cases, the residual functions should return with some way to signal that something went wrong. Is that possible? Do you have time next week for a short meeting to discuss options? 

Thanks!

**Comment by AnHeuermann** [comment]
@TeusvanderStelt can use the `simulationJumpBuffer` as a try-catch-construct:

```C
modelica_boolean failed = TRUE;

/* Try */
#if !defined(OMC_EMCC)
    MMC_TRY_INTERNAL(simulationJumpBuffer)
#endif

/* Function that can throw */
(systemData->residualFunc)(&resUserData, x1, f_x1, (int*)&systemData->size);

failed = FALSE;
/* Catch */
#if !defined(OMC_EMCC)
    MMC_CATCH_INTERNAL(simulationJumpBuffer)
#endif

if (failed) {
  /* Handle failure */
}
```

In this specific case the residual function is failing during `getNonlinearEqns()` ([newton_diagnostics.c#L1006](https://github.com/TeusvanderStelt/OpenModelica/blob/newtonDiagnostics/OMCompiler/SimulationRuntime/c/simulation/solver/newton_diagnostics.c#L1006)) and jumps back all the way to [solver_main.c#L517-L519](https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/SimulationRuntime/c/simulation/solver/solver_main.c#L517-L519).

**Comment by casella** [comment]
Thank @AnHeuermann 👍

**Comment by casella** [comment]
This should be now fixed after #13402
